sycl/doc/design/CommandGraph.md

```diff
@@ -132,6 +132,63 @@ in a command-graph, it will perform a blocking wait on the dependencies of the
 command-group first. The user will experience this wait as part of graph
 finalization.
 
+## Graph Partitioning
+
+To handle dependencies from other devices, the graph can be partitioned during
+the finalization process. A partition is a set of one or more nodes intended
+to run on the same device. Each partition instantiates a command-buffer
+(or equivalent) which contains all the commands to be executed on the device.
+Therefore, the partitioning only impacts graphs in the executable state and
+occurs during finalization. Synchronization between partitions is managed
+by the runtime unlike internal partition dependencies that are handled directly
+by the backend.
+
+Since runtime synchronization and multiple command-buffer involves
+extra latency, the implementation ensures to minimize the number of partitions.
+Currently, the creation of a new partition is triggered by a node containing
+an host-task.
+When a host-task is encountered the predecessors of this host-task node
+are assigned to one partition, the host-task is assigned to another partition,
+and the successors are assigned to a third partition as shown below:
+
+![Graph partition illustration.](images/SYCL-Graph-partitions.jpg)
+
+Partition numbers are allocated in order. Hence, the runtime must ensure that
+Partition `n` complete before starting execution of Partition `n+1`.
+
+Note that partitioning can only happen during the finalization stage due to
+potential backward dependencies that could be created using
+the `make_edge` function.
+
+### Example
+The partitioning process is achieved is two main stages:
+
+1 - Nodes are assigned to a temporary group/partition.
+2 - Once all the nodes have been annotated with a group number, 
+actual partitions are created based on these annotations.
```

Comment:
Sorry I didn't get this suggestion right the first time, need a blank line to avoid the rendered view sticking this two points on the same line.
```suggestion
1 - Nodes are assigned to a temporary group/partition.

2 - Once all the nodes have been annotated with a group number, 
actual partitions are created based on these annotations.
```